cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -50,13 +48,19 @@ import qualified PlutusTx as Pl (FromData (fromBuiltinData))
 -- 'Plutus.Contract.Contract' monad to generate and submit transactions and to
 -- test the code by running it in the 'Cooked.MockChain.Monad.Direct.MockChain' monad, provided in this library.
 class (MonadFail m) => MonadBlockChain m where
+  type BalanceAndCollateralSource m :: *
+
   -- | Generates and balances a transaction from a skeleton, then attemps to validate such
   --  transaction. A balanced transaction is such that @inputs + mints == outputs + fees@.
   --  To balance a transaction, we need access to the current UTxO state to choose
   --  which inputs to add in case the output-side of the balancing equation is bigger.
   --
   --  The 'TxSkel' receives a 'TxOpts' record with a number of options to customize how validation works.
-  validateTxSkel :: TxSkel -> m Pl.CardanoTx
+  validateTxSkel ::
+    BalanceAndCollateralSource m ->   -- the source of funds for balancing and collateral (will also be a signer)
+    [Wallet] ->                       -- the list of additional signers
```

Comment:
This might also need to be `[BalanceAndCollateralSource m]`, but we don't have the `Contract` monad implementation right now, so right now it's a pretty arbitrary decision.